examples/simple/rexchange.cpp

```diff
@@ -0,0 +1,153 @@
+void  destroy_block(void* b)              { delete static_cast<Block*>(b); }
+void  save_block(const void* b,
+                 diy::BinaryBuffer& bb)   { diy::save(bb, *static_cast<const Block*>(b)); }
+void  load_block(void* b,
+                 diy::BinaryBuffer& bb)   { diy::load(bb, *static_cast<Block*>(b)); }
+
+// enqueue data inside the link
+void enq(Block* b, const diy::Master::ProxyWithLink& cp)
+{
+    diy::Link* l = cp.link();
+
+    for (size_t i = 0; i < l->size(); ++i)
+        cp.enqueue(l->target(i), b->count);
+    b->count++;
+}
+
+// dequeue data inside the link
+void deq(Block* b, const diy::Master::ProxyWithLink& cp)
+{
+    diy::Link* l = cp.link();
+
+    for (size_t i = 0; i < l->size(); ++i)
+    {
+        int gid = l->target(i).gid;
+        if (cp.incoming(gid).size())
+        {
+            cp.dequeue(cp.link()->target(i).gid, b->count);
+            fmt::print(stderr, "Dequeue: gid {} received value {} from link gid {}\n",
+                    cp.gid(), b->count, l->target(i).gid);
+            b->count++;
+        }
+    }
+}
+
+// enqueue remote data
+// there is still a link, but you can send to any BlockID = (gid, proc)
+void remote_enq(
+        Block*                              b,
+        const diy::Master::ProxyWithLink&   cp,
+        const diy::Assigner&                assigner)
+{
+    // as a test, send my gid to block 2 gids away (in a ring), which is outside the link
+    // (the link has only adjacent block gids)
+    int my_gid              = cp.gid();
+    int dest_gid            = (my_gid + 2) % assigner.nblocks();
+    int dest_proc           = assigner.rank(dest_gid);
+    diy::BlockID dest_block = {dest_gid, dest_proc};
+    cp.enqueue(dest_block, my_gid);
+}
+
+// dequeue remote data
+// there is still a link, but exchange(remote = true) exchanged messages from any block
+void remote_deq(Block* b, const diy::Master::ProxyWithLink& cp)
+{
+    std::vector<int> incoming_gids;
+    cp.incoming(incoming_gids);
+    for (size_t i = 0; i < incoming_gids.size(); i++)
+        if (cp.incoming(incoming_gids[i]).size())
+        {
+            int recvd_data;
+            cp.dequeue(incoming_gids[i], recvd_data);
+            fmt::print(stderr, "Remote dequeue: gid {} received value {} from gid {}\n",
+                    cp.gid(), recvd_data, incoming_gids[i]);
+        }
+}
+
+int main(int argc, char* argv[])
+{
+//     diy::create_logger("trace");
+
+    diy::mpi::environment     env(argc, argv);
+    diy::mpi::communicator    world;
+    diy::FileStorage          storage("./DIY.XXXXXX");
+    int                       nblocks    = 24 * world.size();
+//     int                       threads    = 1;
+    int                       threads    = 2;
+    int                       mem_blocks = -1;
+
+    diy::Master               master(world,
+                                     threads,
+                                     mem_blocks,
+                                     &create_block,
```

Comment:
We could simplify this example and get rid of `{create,destroy,save,load}_block` and `storage`. They are never used in this example.